app/transactions/views.py

```diff
@@ -47,6 +45,28 @@ def get_context_data(self, **kwargs: Any) -> Dict[str, Any]:
             "amount__count"
         ]
         context["transactions"] = transactions.order_by("-date")
+        self.map_filtered_transactions_to_context("debit", context)
+        self.map_filtered_transactions_to_context("credit", context)
+
+        return context
+
+    def map_filtered_transactions_to_context(
+        self, trans_type: str, context: dict[str, Any]
+    ) -> Dict[str, Any]:
+        filtered_transactions = Transaction.objects.filter(transaction_type=trans_type)
```

Comment:
This starts a blank query that wont be applying any filters for `payee` or `account`.

This would mean that if you click the link for a payee on the transaction view to get a list of transactions filtered payee - the stats table at the stop woudl still be calculating summary stats for all transactions from all Payees

I also realised its super inefficient hitting the DB 3 times. Once to list all transactions and calculate their stats

A second time to get stats only for debits
A third time to get stats only for credits.

Since the 2nd and 3rd queries return a subset of the data already returned by the first first we shouldn't need to do 3 queries. 

We should be able to query the DB once to get both debit and credit transactions and then calculate the stats ourselves.

```python
    def get_transaction_stats(self, transactions):
        if not transactions:
            return {"total": 0, "mean": 0, "count": 0}
        total = sum(t.amount for t in transactions)
        mean = total / len(transactions)
        count = len(transactions)
        return {"total": total, "mean": mean, "count": count}
```

Calculate stats for all returned transactions
```python
        context["aggregate_stats"] = self.get_transaction_stats(transactions)
```

Calculate stats only for debits (where money is coming in)
```python
        context["debit_stats"] = self.get_transaction_stats(
            [t for t in transactions if t.amount < 0]
        )
```

Reply:
The best option to test it works properly is to go to the transactions page and click a payee link to filter by that payee.

If you filter by a payee for rent or utilities or like netflix it becomes very apparent whether it is working properly.